test/unit/test_spike_source.cpp

```diff
@@ -115,3 +115,66 @@ TEST(spike_source, exhaust)
     test_seq(regular_schedule(0, 1, 5));
     test_seq(explicit_schedule({0.3, 2.3, 4.7}));
 }
+
+TEST(spike_source, multiple)
+{
+    // This test assumes that seq will exhaust itself before t=10 ms.
+    auto test_seq = [](std::vector<schedule> seq) {
```

Comment:
you could probably simplify and combine the two tests by having a variadic lambda using universal references (you could then also test for r-value references in the same test)
```
auto test_seq = [](auto&&... seqs) {
    // ...
    spike_source_cell("src", static_cast<decltype(seqs)&&>(seqs)...);
    // ...
}
```

Reply:
Good point, the split was more based on my internal organisation of matters than anything else!

Reply:
On second thought though
1. constructing the `expected` value inside the lambda is kind of awkward now
   ```cxx
           std::vector<schedule> schedules{seqs...};
        for (auto& schedule: schedules) {
            auto ts = as_vector(schedule.events(0, 10));
            actual.insert(actual.end(),
                          ts.begin(), ts.end());
        }
        ```
        It works though since once we invoke the copy ctr and the other time it's the init list, but
        I find that kind of confusing to read.
2. We now have to explicitly cast to a std::vector<...> when calling `test_seq`
3. `static_cast<decltype(seqs)&&>` this is not needed.